astropy/modeling/functional_models.py

```diff
@@ -610,18 +610,12 @@ class Sersic1D(Fittable1DModel):
     amplitude = Parameter(default=1)
     r_eff = Parameter(default=1)
     n = Parameter(default=4)
-    _gammaincinv = None
 
-    def __init__(self, amplitude=amplitude.default, r_eff=r_eff.default,
-                 n=n.default, **kwargs):
-        try:
-            from scipy.special import gammaincinv
-            self.__class__._gammaincinv = gammaincinv
-        except ValueError:
-            raise ImportError("Sersic1D model requires scipy > 0.11.")
-
-        super(Sersic1D, self).__init__(
-            amplitude=amplitude, r_eff=r_eff, n=n, **kwargs)
+    try:
+        from scipy.special import gammaincinv
+        _gammaincinv = gammaincinv
+    except ValueError:
```

Comment:
Ah, yes, thanks.

Setting the attributes in the `__init__` does't work with compound models.  So, we could catch the `scipy` `ImportError` and in that case set the attributes to `None` at the class level (so the module with import without `scipy`), but then re-raise `ImportError` in the `evaluate` method if the attribute is `None`.  Does that sound OK?

The other option is to go back to the previous method (a couple iterations back) of setting these necessary values as module-level constants.  I don't like that though since they will be computed unnecessarily at module import and only a few models need them.